hevm-cli/hevm-cli.hs

```diff
@@ -308,18 +309,29 @@ equivalence cmd = do
                           , rpcInfo = Nothing
                           }
 
-  withSolvers Z3 3 Nothing $ \s -> do
+  solver <- getSolver cmd
+  withSolvers solver 3 Nothing $ \s -> do
     res <- equivalenceCheck s bytecodeA bytecodeB veriOpts Nothing
-    case not (any isCex res) of
+    case any isCex res of
       False -> do
         putStrLn "No discrepancies found"
         when (any isTimeout res) $ do
           putStrLn "But timeout(s) occurred"
           exitFailure
       True -> do
-        putStrLn $ "Not equivalent. Counterexample(s):" <> show res
+        putStrLn $ "Not equivalent. Counterexample(s):" <> show (filter isQed res)
```

Comment:
This will only keep the QEDs in the list. Shouldn't we show the CEXs?

Reply:
Good catch!